src/fsharp/FSharp.Core/option.fs

```diff
@@ -31,15 +43,39 @@ namespace Microsoft.FSharp.Core
         [<CompiledName("ForAll")>]
         let forall p inp = match inp with None -> true | Some x -> p x
 
+        [<CompiledName("Contains")>]
+        let contains x inp = match inp with None -> false | Some v -> v = x
```

Comment:
I've been considering what the relative benefits of `inline` are to letting JIT handle it. I've posted a gist with the IL that was generated on my computer for the [Option module](https://gist.github.com/neoeinstein/532fe7b155f3d97fc0fc2ed2f2f62c68#file-optionmodule_decompiled-cs-L427). Most of the functions are less than 32 bytes of IL, with the only exceptions being `get`, `map2, and `map3`, which makes them largely eligible for JIT-inlining. What incremental benefit does the `inline` keyword provide in this case?